client-app/src/Components/DonatedItemDetails.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import PersonIcon from '@mui/icons-material/Person';
+import CategoryIcon from '@mui/icons-material/Category';
+import EventNoteIcon from '@mui/icons-material/EventNote';
+import AssignmentTurnedInIcon from '@mui/icons-material/AssignmentTurnedIn';
+import '../css/DonatedItemDetails.css'; // Import the new CSS file
+
+interface Donor {
+    id: number;
+    firstName: string;
+    lastName: string;
+    contact: string;
+    email: string;
+    addressLine1: string;
+    addressLine2?: string;
+    state: string;
+    city: string;
+    zipcode: string;
+    emailOptIn: boolean;
+}
+
+interface Program {
+    id: number;
+    name: string;
+    description: string;
+    startDate: string;
+    aimAndCause: string;
+}
+
+interface StatusLog {
+    id: number;
+    type: string;
+    dateModified: string;
+}
+
+interface DonatedItem {
+    id: number;
+    itemType: string;
+    currentStatus: string;
+    dateDonated: string;
+    lastUpdated: string;
+    donor: Donor;
+    program: Program;
+    statuses: StatusLog[];
+}
```

Comment:
@saikiran0405 Move all these interfaces into modals folder, so that they can be reused in multiple places